python/cudf/dataframe/index.py

```diff
@@ -205,7 +206,14 @@ def __ge__(self, other):
     def equals(self, other):
         if len(self) != len(other):
             return False
-        return (self == other)._values.all()
+        elif len(self) == 1:
+            return self[0] == other[0]
+        else:
+            result = (self == other)
+            if isinstance(result, bool):
+                return result
+            else:
+                return (self == other)._values.all()
```

Comment:
You're running the binaryop twice unnecessarily here.
```suggestion
                return result._values.all()
```

Reply:
👍